src/diracx/db/jobs/db.py

```diff
@@ -405,3 +447,298 @@ async def get_wms_time_stamps(self, job_id):
+
+        # Generate groups of TQs that will have the same prio=sum(prios) maomenos
+        rows = await self.retrieve_task_queues(list(tqDict))
+        # TODO: check the following asumption is correct
+        allTQsData = rows
+        tqGroups: dict[str, list] = {}
+        for tqid in allTQsData:
+            tqData = allTQsData[tqid]
+            for field in ("Jobs", "Priority") + priorityIgnoredFields:
+                if field in tqData:
+                    tqData.pop(field)
+            tqHash = []
+            for f in sorted(tqData):
+                tqHash.append(f"{f}:{tqData[f]}")
+            tqHash = "|".join(tqHash)
+            if tqHash not in tqGroups:
+                tqGroups[tqHash] = []
+            tqGroups[tqHash].append(tqid)
+        groups = [tqGroups[td] for td in tqGroups]
+
+        # Do the grouping
+        for tqGroup in groups:
+            totalPrio = 0
+            if len(tqGroup) < 2:
+                continue
+            for tqid in tqGroup:
+                totalPrio += tqDict[tqid]
+            for tqid in tqGroup:
+                tqDict[tqid] = totalPrio
+
+        # Group by priorities
+        prioDict: dict[int, list] = {}
+        for tqId in tqDict:
+            prio = tqDict[tqId]
+            if prio not in prioDict:
+                prioDict[prio] = []
+            prioDict[prio].append(tqId)
+
+        # Execute updates
+        for prio, tqs in prioDict.items():
+            update_stmt = (
+                update(TaskQueues).where(TaskQueues.TQId.in_(tqs)).values(Priority=prio)
+            )
+            await self.conn.execute(update_stmt)
+
+    async def retrieve_task_queues(self, tqIdList=None):
+        """
+        Get all the task queues
+        """
+        if tqIdList is not None and not tqIdList:
+            # Empty list => Fast-track no matches
+            return {}
+
+        stmt = (
+            select(
+                TaskQueues.TQId,
+                TaskQueues.Priority,
+                func.count(JobsQueue.TQId).label("Jobs"),
+                TaskQueues.Owner,
+                TaskQueues.OwnerGroup,
+                TaskQueues.VO,
+                TaskQueues.CPUTime,
+            )
+            .select_from(TaskQueues.join(JobsQueue, TaskQueues.TQId == JobsQueue.TQId))
+            .select_from(
+                TaskQueues.join(SitesQueue, TaskQueues.TQId == SitesQueue.TQId)
+            )
+            .select_from(
+                TaskQueues.join(GridCEsQueue, TaskQueues.TQId == GridCEsQueue.TQId)
+            )
+            .group_by(
+                TaskQueues.TQId,
+                TaskQueues.Priority,
+                TaskQueues.Owner,
+                TaskQueues.OwnerGroup,
+                TaskQueues.VO,
+                TaskQueues.CPUTime,
+            )
+        )
+        if tqIdList is not None:
+            stmt = stmt.where(TaskQueues.TQId.in_(tqIdList))
+
+        tqData = dict(row._mapping for row in await self.conn.execute(stmt))
```

Comment:
I think you probably want this:
```suggestion
        tqData = dict(dict(row._mapping) for row in await self.conn.execute(stmt))
```